### PR TITLE
fix: tests were overwriting real settings.json

### PR DESCRIPTION
**Pull request by soyelmismo**
## Problem

Two related issues:

### 1. Tests were overwriting real settings.json

`resolve_api_key()` called `load_settings()` + `save_settings()` internally. Tests that invoked it with a CLI key (e.g. `"sk-from-cli"`) silently persisted to the user's real `~/.config/tinyharness/settings.json`, corrupting their config.

### 2. Settings deserializer too strict

When users upgraded from a build that adds new fields (e.g. the `auto_accept_mode` enum), their existing settings.json didn't have every field the current `Settings` struct expects. This caused noisy 'parse error: missing field auto_accept_safe_commands' warnings on every launch and silently fell back to defaults, losing user preferences.

### Settings
- Add `#[serde(default)]` at the struct level so missing fields use per-field defaults (ProviderKind → Ollama, AgentMode → Casual, OllamaThinkType → Medium, auto_accept_safe_commands → true).
- Derive `Default` on `ProviderKind`, `AgentMode`, `OllamaThinkType`.
- Add unit tests covering both empty-object and missing-field cases.

### Test isolation
Split `resolve_api_key` into two functions:
- `resolve_api_key(cli_api_key, settings)` — production, with side-effects.
- `resolve_api_key_pure(cli_api_key, settings)` — pure, no I/O. Used by all 5 tests.

The production version also now skips the `load_settings + save_settings` round-trip when the CLI key is empty.

## Verification

```
Before: md5sum of settings.json = X
cargo test → 89 + 77 + 325 + 2 = 493 passed
After:  md5sum of settings.json = X  (identical)
```

Tests pass on Linux + clippy clean + fmt clean.

### CODE REVIEW

File: tinyharness-lib/src/config/mod.rs [hunk 1]
```diff
@@ -711,3 +723,52 @@ pub fn ensure_prompts_initialized() -> PathBuf {
...
+    use super::*;
+
+    /// User settings from a future build (with `auto_accept_mode`) must load
+    /// against the current `Settings` struct (which still uses
+    /// `auto_accept_safe_commands`) without parse errors.
+    #[test]
+    fn load_settings_missing_fields_uses_defaults() {
+        let json = r#"{
+            "last_provider": "OpenAiCompat",
+            "last_provider_url": "http://localhost:8080/v1",
```
Comment (PTFOPlayer) [inline tinyharness-lib/src/config/mod.rs:740]: Please use localhost or api.example.com as test url

**Review by PTFOPlayer — APPROVED** [review]
